src/main/java/com/google/swarm/tokenization/common/FileReaderTransform.java

```diff
@@ -60,27 +79,76 @@ public PCollectionTuple expand(PBegin input) {
+        .apply("ReadFile", ParDo.of(new FileReaderSplitDoFn(batchSize())));
   }
 
   public class MapPubSubMessage extends DoFn<PubsubMessage, String> {
+    private static final String VALID_FILE_PATTERNS = "^(.(?!.*\\.ctl$|.*\\.dlp$|.*\\.xml$|.*\\.json$|.*parallel_composite_uploads.*$|.*\\.schema$|.*\\.temp$))*$";
+
+    private final Counter numberOfFilesReceived =
+            Metrics.counter(FileReaderTransform.MapPubSubMessage.class, "NumberOfFilesReceived");
 
     @ProcessElement
     public void processElement(ProcessContext c) {
+
+      LOG.info("File Read Transform:ConvertToGCSUri: Located File's Metadata : "+c.element().getAttributeMap());
+      numberOfFilesReceived.inc(1L);
+
       String bucket = c.element().getAttribute("bucketId");
       String object = c.element().getAttribute("objectId");
       String eventType = c.element().getAttribute("eventType");
+      String file_ts_string = c.element().getAttribute("eventTime");
       GcsPath uri = GcsPath.fromComponents(bucket, object);
 
-      if (eventType.equalsIgnoreCase(Util.ALLOWED_NOTIFICATION_EVENT_TYPE)) {
-        LOG.info("File Name {}", uri.toString());
-        c.output(uri.toString());
-      } else {
-        LOG.info("Event Type Not Supported {}", eventType);
-      }
+      String file_name = uri.toString();
+      String prefix;
+
+      //Match filenames having extensions
+      Matcher m1 = Pattern.compile("^gs://([^/]+)/(.*)\\.(.*)$").matcher(file_name);
+
+            if (m1.find()) {
+                prefix = m1.group(2);
+            } else {//No extension
+                prefix = object;
+            }
+
+            ImmutableList.Builder<ResourceId> sourceFiles = ImmutableList.builder();
+            AtomicBoolean should_scan = new AtomicBoolean(true);
+
+                if (!file_name.matches(VALID_FILE_PATTERNS)) {
+                    LOG.warn("File Read Transform:ConvertToGCSUri: Unsupported File Format. Skipping: {}", file_name);
+                    should_scan.set(false);
+                } else if (!eventType.equalsIgnoreCase(Util.ALLOWED_NOTIFICATION_EVENT_TYPE)) {
+                    LOG.warn("File Read Transform:ConvertToGCSUri: Event Type Not Supported: {}. Skipping: {}", eventType,file_name);
+                    should_scan.set(false);
+                } else {
+                    try {
+                        MatchResult listResult = FileSystems.match("gs://" + bucket + "/" + prefix + ".*.dlp", EmptyMatchTreatment.ALLOW);
+                        listResult.metadata().forEach(metadata -> {
+                            ResourceId resourceId = metadata.resourceId();
+                            Instant file_ts = Instant.parse(file_ts_string);
+                            Instant tf_ts = new Instant(metadata.lastModifiedMillis());
+                            LOG.warn(file_ts.toString());
+                            LOG.warn(tf_ts.toString());
```

Comment:
Should have one statement and a meaning error message. This is not helpful. I feel you have too many logs here. Let's try to summarize and output what will be useful

Reply:
Yes sure. Some parts of the code are added by Robert L. Will look into those parts and take out the unnecessary logs.